packages/Angular/Explorer/explorer-core/src/lib/resource-browser/resource-browser.component.ts

```diff
@@ -77,19 +80,7 @@ export class ResourceBrowserComponent {
   /**
    * Array of NewItemOption objects that will be displayed in the Create New dropdown. Defaults to having a single entry for creating a new folder, you can remove this or add to it.
    */
-  @Input() NewItemOptions: NewItemOption[] = [{
-    Text: 'New Folder',
-    Description: 'Create a new folder',
-    Icon: 'folder',
-    Action: () => {
-      console.log('New Folder');
-    }
-  }];
-
-  /**
-   * If set, this will set the current Category/Folder for the display. Not all Resource Types support Categories, so if the Resource Type in question does not have a CategoryEntityID specified, this property will be ignored.
-   */
-  @Input() CurrentCategoryID?: string;
+  @Input() NewItemOptions: NewItemOption[] = [];
```

Comment:
@JS-BC  so we start out with an empty array and all users of the class have to add this?

Reply:
@AN-BC we have a private `_DefaultNewItemOptions` property that has the "Create Folder" action. The `Items` property, which is what is hooked up to the dropdown,  gets populated by combining the items the user passes in and the default items. If `EnableCategories` is false then the default items is not added to the Items property